### PR TITLE
chore(test): add unit test for mem table spill

### PR DESCRIPTION
**Pull request by wcy-fdu**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/common/table/test_state_table.rs [hunk 1]
```diff
@@ -2045,3 +2045,133 @@ async fn test_state_table_iter_prefix_and_sub_range() {
     let res = iter.next().await;
     assert!(res.is_none());
 }
+
+#[tokio::test]
+async fn test_mem_table_spill_in_streaming() {
```
Comment (hzxa21) [inline src/stream/src/common/table/test_state_table.rs:2050]: This UT tests the case when there are 3 rows inserted into memtable and `try_flush` is triggered in between. This does test whether the flushed memtable is visible to `get` but doesn't test the spill epoch comparison and whether the latest row with the same key can be retrieved.

In addition to this test, I suggest we add a test or extend the current test to test the following case:
1. insert `(k1, text1)` and `try_flush`
2. insert `(k1, text1')` and `try_flush`
3. get `k1` before and after 2's `try_flush` and make sure `text1'` is returned

File: src/stream/src/common/table/test_storage_table.rs [hunk 2]
```diff
@@ -468,3 +468,196 @@ async fn test_batch_scan_with_value_indices() {
     let res = iter.next_row().await.unwrap();
     assert!(res.is_none());
 }
+
+#[tokio::test]
+async fn test_mem_table_spill_in_batch_mode() {
```
Comment (hzxa21) [inline src/stream/src/common/table/test_storage_table.rs:473]: Same here. It is better to test the muli-version key with different spilled epochs as well.

File: src/stream/src/common/table/test_state_table.rs (hunk 1 above)
Comment (wcy-fdu) [reply src/stream/src/common/table/test_state_table.rs:2050]: I see, supplied.

**Review by wenym1 — COMMENTED** [review]
Generally LGTM. 

But I think `#[cfg(feature = "enable_test_epoch")]` is still enabled in the unit test of streaming? If so, can we still test spill? When it is enabled, we are not able to include spill offset in `EpochWithGap`.

**Comment by wcy-fdu** [comment]
> Generally LGTM.
> 
> But I think `#[cfg(feature = "enable_test_epoch")]` is still enabled in the unit test of streaming? If so, can we still test spill? When it is enabled, we are not able to include spill offset in `EpochWithGap`.

Solved.

File: Cargo.toml [hunk 3]
```diff
@@ -38,6 +38,7 @@ members = [
   "src/storage/hummock_sdk",
   "src/storage/hummock_test",
   "src/stream",
+  "src/stream/mem_table_test",
```
Comment (wenym1) [inline Cargo.toml:41]: Better to rename to `spill_test`?

**Review by wenym1 — APPROVED** [review]

**Review by hzxa21 — APPROVED** [review]
LGTM

**Pull request merged**